internal/provider/database/resource.go

```diff
@@ -64,7 +76,44 @@ func (state *DatabaseResourceModel) Read(ctx context.Context, client openapi.Cli
 	return helper.ParseResponse(resp, state)
 }
 
-func (state *DatabaseResourceModel) Update(ctx context.Context, client openapi.ClientInterface) error {
+const (
+	DBA_PASSWORD_CHANGE_UNSUPPORTED_MSG = "Configured DBA password was changed and the server does not support updating the DBA password. " +
+		"Revert the configured DBA password to the value in the Terraform state and retry."
+)
+
+func IsDbaPasswordUpdateUnsupportedError(resp *http.Response, err error) bool {
+	if err != nil {
+		// "404 Not Found" is returned with no "detail" message if /dbaPassword	sub-resource is not supported
+		if resp.StatusCode == http.StatusNotFound {
+			if apiError, ok := err.(*helper.ApiError); !ok || apiError.GetDetail() == "" {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func (state *DatabaseResourceModel) Update(ctx context.Context, client openapi.ClientInterface, currentState framework.ResourceState) error {
+	// Try to update DBA password if it was changed in config
+	currentDatabase, _ := currentState.(*DatabaseResourceModel)
+	if !state.DbaPasswordMatches(currentDatabase) {
+		request := openapi.UpdateDbaPasswordModel{
+			Current: *currentDatabase.DbaPassword,
+			Target:  state.DbaPassword,
+		}
+		resp, err := client.UpdateDbaPassword(ctx, state.Organization, state.Project, state.Name, nil, request)
```

Comment:
Since we do not specify a timeout, this is not waiting for a password rotation and never checking that the update actually occurred, right? Even if the password change fails, we return a success and don't retry it on subsequent applies.

Reply:
I'm assuming that when you say "password change fails", you're talking about the scenario where the current and target credentials are both incorrect and the `target` entry is removed from the Secret resource to signal that the password change was aborted. Any other failure should the leave the Secret resource in the same state, which should cause the password change to appear as pending.

If the password change failed as I described above, then that would mean that the current password was not actually correct. So there would be no point in retrying the DBA password update with that password as the current password. That is a misconfiguration that would have been caused by someone directly changing the DBA password via SQL and requires a manual intervention.

The main issue with not waiting is that the user may falsely believe that the password change has taken effect and try to use the new DBA password. I considered adding a new database state to signal that a password change is pending, and that would cause the readiness check that is performed after updating the database to also wait until the password has been updated (or aborted, as in the scenario described above).

Reply:
Still, we should signal to the user that the password change failed (so they know that they need to go fix it). 

Manually changing the password to the new password is probably just as easy (or hard) as reverting a manual password change, so my objection to updating the terraform state without making the corresponding change to the actual state is more of a philosophical one.

Reply:
I'll look into adding a password check to the `DatabaseResourceModel.CheckReady()` method.